### PR TITLE
fix: use Envoy's default for validate_clusters to fix breaking routes when some backend clusters don't exist

### PR DESCRIPTION
**Pull request by ndhanushkodi**
### Description

The validate_clusters option in Envoy's [route configuration](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto#config-route-v3-routeconfiguration) says:

"An optional boolean that specifies whether the clusters that the route table refers to will be validated by the cluster manager. If set to true and a route refers to a non-existent cluster, the route table will not load. If set to false and a route refers to a non-existent cluster, the route table will load and the router filter will return a 404 if the route is selected at runtime. This setting defaults to true if the route table is statically defined via the [route_config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-route-config) option. This setting default to false if the route table is loaded dynamically via the [rds](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-rds) option. Users may wish to override the default behavior in certain cases (for example when using CDS with a static route table)."

We are setting it dynamically via RDS, but overriding the default value to set it explicitly to true. This means when a cluster that the route is supposed to point to doesn't exist, the route can fail to route to any of its backends. This case can be triggered if you have a router -> resolver where the resolver has backends on different peers/wan federated backends, and you add a route to a backend that doesn't exist. The non-existent backend causes the existing backends to fail. I was not able to trigger this case in a single cluster setup, but with a peered backend it can be triggered. 

Because, the traffic doesn't just blackhole, but rather returns a 503, this actually seems to be the desired behavior, rather than making all other routing paths within that route fail due to a missing cluster. This is similar to the conclusion that was reached within the [Jira](https://hashicorp.atlassian.net/browse/NET-10435) ticket.

This PR removes the code that overrides the default value of this validate_clusters option. 

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

**Review by zalimeni — COMMENTED** [review]
Initial review LGTM! Once we introduce a flag I'm guessing we'll want to have a golden smoke test to prove it out but the general shape of these changes makes sense to me 👍🏻

**Review by zalimeni — COMMENTED** [review]
Same [comment](https://github.com/hashicorp/consul/pull/21587#pullrequestreview-2242645474) as CE, generally LGTM pending intro of a revert flag and a test for that! 🎉

File: agent/structs/config_entry_mesh.go [hunk 1]
```diff
@@ -20,6 +20,11 @@ type MeshConfigEntry struct {
 	// MutualTLSMode=permissive in either service-defaults or proxy-defaults.
 	AllowEnablingPermissiveMutualTLS bool `json:",omitempty" alias:"allow_enabling_permissive_mutual_tls"`
 
+	// ValidateClusters is false by default and configures whether Envoy proxies will validate clusters in a route. If
+	// set to true and any clusters in the route do not exist, the route table will not load. If set to false, the route
+	// table will load and routing to a non-existent cluster will result in a 404.
```
Comment (zalimeni) [inline agent/structs/config_entry_mesh.go:25]: ~ We might want to add a link the Envoy docs for this option as well
```suggestion
	// table will load and routing to a non-existent cluster will result in a 404.
	// See [Envoy docs] for more details.
	//
	// [Envoy docs]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto#envoy-v3-api-field-config-route-v3-routeconfiguration-validate-clusters
```

File: agent/xds/routes.go [hunk 2]
```diff
@@ -276,18 +285,19 @@ func (s *ResourceGenerator) routesForMeshGateway(cfgSnap *proxycfg.ConfigSnapsho
...
 			VirtualHosts: []*envoy_route_v3.VirtualHost{virtualHost},
-			// ValidateClusters defaults to true when defined statically and false
-			// when done via RDS. Re-set the reasonable value of true to prevent
-			// null-routing traffic.
-			ValidateClusters: response.MakeBoolValue(true),
+		}
+		if mesh := cfgSnap.MeshConfig(); mesh != nil {
+			if mesh.ValidateClusters {
+				route.ValidateClusters = response.MakeBoolValue(true)
+			}
```
Comment (zalimeni) [inline agent/xds/routes.go:292]: Could we use the helper here as well?

File: api/config_entry_mesh.go [hunk 3]
```diff
@@ -26,6 +26,11 @@ type MeshConfigEntry struct {
 	// MutualTLSMode=permissive in either service-defaults or proxy-defaults.
 	AllowEnablingPermissiveMutualTLS bool `json:",omitempty" alias:"allow_enabling_permissive_mutual_tls"`
 
+	// ValidateClusters is false by default and configures whether Envoy proxies will validate clusters in a route. If
+	// set to true and any clusters in the route do not exist, the route table will not load. If set to false, the route
+	// table will load and routing to a non-existent cluster will result in a 404.
```
Comment (zalimeni) [inline api/config_entry_mesh.go:31]: Same comment here re: Envoy doc link

**Review by zalimeni — APPROVED** [review]
Overall LGTM, just a few small comments! Approved to unblock

**Review by zalimeni — COMMENTED** [review]
Oh, one more question @ndhanushkodi , though happy to do this in a follow-up PR: I'm assuming we need some public docs updates as well? Just realized these changes are exlusively go/proto docs.

File: agent/xds/routes.go (hunk 2 above)
Comment (ndhanushkodi) [reply agent/xds/routes.go:292]: Yup!

Comment (ndhanushkodi) [reply agent/xds/routes.go:292]: thanks for catching

**Comment by ndhanushkodi** [comment]
@zalimeni I originally thought to add docs as a followup but it was easy enough so I just went ahead and added here.

**Pull request merged**